### PR TITLE
Implement sum_duplicates

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented `sum_duplicates` for `coo_matrix`. Its bottleneck is `cumsum`, but we can make more efficient implementation:
http://people.cs.vt.edu/yongcao/teaching/cs5234/spring2013/slides/Lecture10.pdf
merge #328 first.
related to #36

### CODE REVIEW

**Comment by okuta** [comment]
How about using thrust?
https://thrust.github.io/doc/group__prefixsums.html

**Comment by unnonouno** [comment]
Wow. So, we can use it in cumsum, right?

**Comment by okuta** [comment]
Yes we can.

**Comment by takagi** [comment]
Would you rebase or merge the master? This PR seems to contain the change in #328 .

File: tests/cupy_tests/sparse_tests/test_coo.py [hunk 1]
```diff
@@ -294,6 +309,44 @@ def test_transpose(self, xp, sp):
...
+        return m.toarray()
+
+    @testing.numpy_cupy_allclose(sp_name='sp')
+    def test_sum_duplicates_empty(self, xp, sp):
+        m = _make_empty(xp, sp, self.dtype)
+        self.assertFalse(m.has_canonical_format)
+        m.sum_duplicates()
+        self.assertTrue(m.has_canonical_format)
+        self.assertEqual(m.nnz, 0)
+        a = m.toarray()
```
Comment (takagi) [inline tests/cupy_tests/sparse_tests/test_coo.py]: `return m.toarray()` would be enough.

**Comment by unnonouno** [comment]
I fixed it.

**Pull request merged**